### PR TITLE
Add network checks

### PR DESCRIPTION
**Pull request by boyuan-chen**
Two new checks are added for the erigon.

### CODE REVIEW

**Review by mmontour1306 — APPROVED** [review]
Looks fine, but a style note if we're trying to match upstream - their op-geth params/config.go has the IsOptimismPreBedrock() after the IsOptimismRegolith() function, and they call IsOptimism() rather than testing c.Optimism directly. 

```
// IsBedrock returns whether num is either equal to the Bedrock fork block or greater.
func (c *ChainConfig) IsBedrock(num *big.Int) bool {
	return isBlockForked(c.BedrockBlock, num)
}

func (c *ChainConfig) IsRegolith(time uint64) bool {
	return isTimestampForked(c.RegolithTime, time)
}

// IsOptimism returns whether the node is an optimism node or not.
func (c *ChainConfig) IsOptimism() bool {
	return c.Optimism != nil
}

// IsOptimismBedrock returns true iff this is an optimism node & bedrock is active
func (c *ChainConfig) IsOptimismBedrock(num *big.Int) bool {
	return c.IsOptimism() && c.IsBedrock(num)
}

func (c *ChainConfig) IsOptimismRegolith(time uint64) bool {
	return c.IsOptimism() && c.IsRegolith(time)
}

// IsOptimismPreBedrock returns true iff this is an optimism node & bedrock is not yet active
func (c *ChainConfig) IsOptimismPreBedrock(num *big.Int) bool {
	return c.IsOptimism() && !c.IsBedrock(num)
}


```

File: types/txn.go [hunk 1]
```diff
@@ -334,22 +334,37 @@ func (ctx *TxParseContext) parseTransactionBody(payload []byte, pos, p0 int, slo
...
 		dataPos, dataLen, err = rlp.String(payload, p) // Value
+		if err != nil {
+			log.Warn("failed to parse value", "err", err)
+		}
 		p = dataPos + dataLen
 
 		p, slot.Gas, err = rlp.U64(payload, p)
 		if err != nil {
-			return 0, fmt.Errorf("%w: d_gas: %s", ErrParseTxn, err)
+			return 0, fmt.Errorf("%w: d_gas: %s", ErrParseTxn, err) //nolint
```
Comment (jyellick) [inline types/txn.go:367]: I'm curious what the linting error is here / whether we could just fix it?

**Review by jyellick — APPROVED** [review]
LGTM

File: types/txn.go (hunk 1 above)
Comment (boyuan-chen) [reply types/txn.go:367]: The lint error is
```
types/txn.go:367:55: non-wrapping format verb for fmt.Errorf. Use `%w` to format errors (errorlint)
                        return 0, fmt.Errorf("%w: d_gas: %s", ErrParseTxn, err)
                                                                           ^
make: *** [lint] Error 1
```

**Pull request merged**